api/client/output.proto

```diff
@@ -44,14 +45,30 @@ message Statistic {
   }
 }
 
+// An output generated by a UserDefinedOutput plugin.
+message UserDefinedOutput {
+  // The official name of the plugin as registered through Envoy.
```

Comment:
Sorry, can we clarify this a bit more, as it stands this might be read as there is some Envoy instance the plugins have to register with. But I think what we wanted to say is that the plugins register through the Envoy's plugin registration mechanism.